atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/expectations/KeyValueServiceDataTrackerTest.java

```diff
@@ -0,0 +1,306 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl.expectations;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.transaction.api.expectations.ImmutableKvsCallReadInfo;
+import com.palantir.atlasdb.transaction.api.expectations.ImmutableTransactionReadInfo;
+import com.palantir.atlasdb.transaction.api.expectations.KvsCallReadInfo;
+import com.palantir.atlasdb.transaction.api.expectations.TransactionReadInfo;
+import com.palantir.common.concurrent.PTExecutors;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import org.junit.Test;
+
+public final class KeyValueServiceDataTrackerTest {
+    private static final TableReference TABLE_1 = TableReference.createWithEmptyNamespace("Table1");
+    private static final TableReference TABLE_2 = TableReference.createWithEmptyNamespace("Table2");
+    private static final TableReference TABLE_3 = TableReference.createWithEmptyNamespace("Table3");
+    private static final String KVS_METHOD_NAME_1 = "getRows";
+    private static final String KVS_METHOD_NAME_2 = "getAsync";
+    private static final String KVS_METHOD_NAME_3 = "get";
+    private static final String KVS_METHOD_NAME_4 = "getMetadataForTables";
+    private static final String KVS_METHOD_NAME_5 = "getMetadataForTable";
+    private static final long NO_BYTES_READ = 0L;
+    private static final long BYTES_READ_1 = 83L;
+    private static final long BYTES_READ_2 = 103L;
+    private static final long BYTES_READ_3 = 971L;
+    private static final int CONCURRENCY_ROUNDS = 20_000;
+
+    private final KeyValueServiceDataTracker tracker = new KeyValueServiceDataTracker();
+
+    @Test
+    public void noReadsTracksNothing() {
+        assertThat(tracker.getReadInfo()).isEqualTo(createTransactionReadInfo(0, 0));
+        assertThat(tracker.getReadInfoByTable()).isEmpty();
+    }
+
+    @Test
+    public void recordReadForTableCallsAreTracked() {
+        tracker.recordReadForTable(TABLE_1, KVS_METHOD_NAME_1, BYTES_READ_1);
```

Comment:
(then either consider doing that for containsExactlyEntriesOf or comment, but it's less necessary since hopefully createTransactionReadInfo signature is made clearer from the variable at L70)